src/warnings.ts

```diff
@@ -28,6 +29,12 @@ export function warn(code: string, message: string) {
   if (isBrowser()) {
     console.warn(message);
   } else {
-    process.emitWarning(message);
+    if (typeof warnType !== 'undefined') {
```

Comment:
nit(style): please merge this into one if-else if-else chain to reduce nesting :)